src/smtml/expr.ml

```diff
@@ -420,11 +425,16 @@ let rec relop ty op hte1 hte2 =
   | Ne, _, Val (App (`Op "symbol", [ Str _ ]))
   | Ne, Val (App (`Op "symbol", [ Str _ ])), _ ->
     value True
+  | Eq, (Symbol s1 as e1), (Symbol s2 as e2) ->
+    if Symbol.equal s1 s2 then
+      make (Unop (Ty_bool, Not, make (Unop (Ty_bool, Is_nan, make e1))))
+    else
+      make (Relop (Ty_fp 32, Eq, make e1, make e2))
```

Comment:
Same here again. Here it looks like you assume that the type is a type parameter and not the type of the resulting expression, which is contrary to the previous case.